File: java/13514022_ITI/src/makhluk/Satelit.java
package makhluk;
/** Kelas 'Satelit'. Turunan (public) dari Planet
 * @author Malvin J./13514044
 * {@inheritDoc}
 */
public class Satelit extends Planet {
    /** Ctor makhluk 'Satelit'.
    * Selain atribut id, posisi, dan gambar, nilai atribut dirandom.
    * Gambar = 'C'.
    *
    * @param id ID
    * @param x absis awal
    * @param y ordinat awal
    * @param time waktu lahir
    *
    */
    public Satelit(final int id, final int x, final int y, final int time) {
        final double kali = 0.2;
        setArah();
        setKekuatan();
        setID(id);
        setStartTime(time);
        setPos(x, y);
        setDTime();
        setGambar('C');
        setGravitasi((int) kali * getKekuatan());
    }
}

File: cpp/2B - NONTHREAD - STL - TOMBOL/Satelit.cpp
/*  Nama: Sashi Novitasari/13514027
    Tanggal: 3/3/2016
    Kelas Makhluk */

#include "Shape.h"
using namespace std;

Satelit::Satelit(int id, int x, int y,int time){
    setArah();
    setKekuatan();
    ID = id;
    starttime = time;
    pos = new Point(x,y);
    setDTime();
    gambar = 'C';
    gravitasi = (int) 0.2*kekuatan;
}

Satelit::~Satelit(){}
